tests/NCronJob.Tests/NCronJobIntegrationTests.cs

```diff
@@ -69,6 +69,23 @@ public async Task EachJobRunHasItsOwnScope()
         storage.Guids.Distinct().Count().ShouldBe(storage.Guids.Count);
     }
 
+    [Fact]
+    public async Task ExecuteAnInstantJobWithoutPreviousRegistration()
+    {
+        ServiceCollection.AddNCronJob();
+
+        var provider = CreateServiceProvider();
+        await provider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
+
+        provider.GetRequiredService<IInstantJobRegistry>().RunInstantJob<SimpleJob>();
```

Comment:
Interesting. Given that test and this comment: https://github.com/NCronJob-Dev/NCronJob/issues/54#issuecomment-2453578026

We are more and more going down the road of utilizing `Activator.CreateInstance`.

Reply:
That test was added on purpose to actually explicitly showcase that this **currently** works.

Once #108 is completely defined (and what should be expected of a jobdefinition, when it should be registered, ....), we'll see more clearly about whether or not we still want/need reflection.

As #108 can't land in the context of v3, stopping reliance on Activator.CreateInstance can be part of the breaking changes.

Reply:
Good way of thinking - i like